tests/trestle/core/remote/cache_test.py

```diff
@@ -0,0 +1,371 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2020 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Testing for cache functionality."""
+
+import os
+import pathlib
+import random
+import string
+from unittest.mock import patch
+
+import pytest
+
+import trestle.core.err as err
+from trestle.core import generators
+from trestle.core.err import TrestleError
+from trestle.core.remote import cache
+from trestle.core.settings import Settings
+from trestle.oscal.catalog import Catalog
+from trestle.utils import fs
+
+
+def test_fetcher_oscal(tmp_trestle_dir):
+    """Test whether fetcher can get an object from the cache as an oscal model."""
+    # Fetch from local content, expecting it to be cached and then fetched as an oscal model.
+    rand_str = ''.join(random.choice(string.ascii_letters) for x in range(16))
+    catalog_file = pathlib.Path(tmp_trestle_dir / f'{rand_str}.json').__str__()
+    catalog_data = generators.generate_sample_model(Catalog)
+    catalog_data.oscal_write(pathlib.Path(catalog_file))
+    fetcher = cache.FetcherFactory.get_fetcher(pathlib.Path(tmp_trestle_dir), catalog_file, False, False)
+    # Create/update the cache copy
+    fetcher._refresh = True
+    fetcher._cache_only = False
+    fetcher._update_cache()
+    fetched_data = fetcher.get_oscal(Catalog)
+    # Make last_modified identical then compare as this alone is expected to differ:
+    fetched_data.metadata.last_modified = catalog_data.metadata.last_modified
+    assert fetched_data == catalog_data
```

Comment:
The localhost fetcher should be restricting this from happening (e.g. the file in the cache is inside the same trestle directory).

Reply:
Fixed. Local fetcher will refuse to update the cache if the uri is inside a trestle project.